Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -67,93 +68,112 @@ static String symbolToString(const CSSValue* value)
+        } else
+            return { };
+    }
+    return result;
+}
+
 static CSSCounterStyleDescriptors::Pad translatePadFromStyleProperties(const StyleProperties& properties)
 {
-    auto pad = properties.getPropertyCSSValue(CSSPropertyPad).get();
-    if (!pad || !is<CSSValueList>(pad))
-        return CSSCounterStyleDescriptors::Pad();
-
-    auto& list = downcast<CSSValueList>(*pad);
-    ASSERT(list.size() == 2);
-    auto length = downcast<CSSPrimitiveValue>(list.item(0));
-    if (!length)
-        return CSSCounterStyleDescriptors::Pad();
-    auto lengthValue = length->intValue();
-    ASSERT(lengthValue >= 0);
-    return { static_cast<unsigned>(std::max(0, lengthValue)), symbolToString(list.item(1)) };
+    auto value = properties.getPropertyCSSValue(CSSPropertyPad);
+    if (!value)
+        return { };
+
+    auto& list = downcast<CSSValueList>(*value);
+    ASSERT(list->size() == 2);
+    auto length = downcast<CSSPrimitiveValue>(list->item(0)).intValue();
+    ASSERT(length >= 0);
+    return { static_cast<unsigned>(std::max(0, length)), symbolToString(list->item(1)) };
 }
 
 static CSSCounterStyleDescriptors::NegativeSymbols translateNegativeSymbolsFromStyleProperties(const StyleProperties& properties)
 {
-    auto negative = properties.getPropertyCSSValue(CSSPropertyNegative).get();
-    if (!negative || !is<CSSValueList>(negative))
-        return CSSCounterStyleDescriptors::NegativeSymbols();
+    auto negative = properties.getPropertyCSSValue(CSSPropertyNegative);
+    if (!negative)
+        return { };
 
-    auto& list = downcast<CSSValueList>(*negative);
-    ASSERT(list.size() == 1 || list.size() == 2);
     CSSCounterStyleDescriptors::NegativeSymbols result;
-    if (list.size() >= 1)
-        result.m_prefix = symbolToString(list.item(0));
-    else if (list.size() == 2)
-        result.m_suffix = symbolToString(list.item(1));
-    else {
-        ASSERT_NOT_REACHED();
-        return CSSCounterStyleDescriptors::NegativeSymbols();
+    if (auto list = dynamicDowncast<CSSValueList>(negative.get())) {
+        if (list->size() >= 1)
+            result.m_prefix = symbolToString(list->item(0));
+        else if (list->size() == 2)
+            result.m_suffix = symbolToString(list->item(1));
+        else {
+            ASSERT_NOT_REACHED();
+            return { };
+        return result;
+        }
     }
+    result.m_prefix = symbolToString(negative.get());
     return result;
 }
 
 static Vector<CSSCounterStyleDescriptors::Symbol> translateSymbolsFromStyleProperties(const StyleProperties& properties)
 {
-    auto symbolsValues = properties.getPropertyCSSValue(CSSPropertySymbols).get();
-    if (!symbolsValues || !is<CSSValueList>(symbolsValues))
+    auto symbolsValues = properties.getPropertyCSSValue(CSSPropertySymbols);
+    if (!symbolsValues)
+        return { };
+    CSSValueList* symbols;
+    if (!(symbols = downcast<CSSValueList>(symbolsValues.get())))
         return { };
```

Comment:
consumeCounterStyleSymbols suggests this is always a list too. The same idiom applies here too.